test/client_mqtt5.js

```diff
@@ -13,29 +13,548 @@ describe('MQTT 5.0', function () {
+            assert.strictEqual(packet.properties.topicAlias, 1)
+            break
+          case 1:
+            assert.strictEqual(packet.topic, '')
+            assert.strictEqual(packet.properties.topicAlias, 1)
+            break
+          case 2:
+            assert.strictEqual(packet.topic, '')
+            assert.strictEqual(packet.properties.topicAlias, 1)
+            server103.close()
+            client.end(true, done)
+            break
+        }
+      })
+    }).listen(ports.PORTAND103)
+
+    client.on('connect', function () {
+      // register topicAlias
+      client.publish('test1', 'Message', { properties: { topicAlias: 1 } })
+      // use topicAlias
+      client.publish('', 'Message', { properties: { topicAlias: 1 } })
+      // use topicAlias by autoApplyTopicAlias
+      client.publish('test1', 'Message')
+    })
+  })
+
+  it('topicAlias is automatically used if autoAssignTopicAlias is true', function (done) {
+    this.timeout(15000)
+
+    var opts = {
+      host: 'localhost',
+      port: ports.PORTAND103,
+      protocolVersion: 5,
+      autoAssignTopicAlias: true
+    }
+    var client = mqtt.connect(opts)
+
+    var publishCount = 0
+    var server103 = new MqttServer(function (serverClient) {
+      serverClient.on('connect', function (packet) {
+        serverClient.connack({
+          reasonCode: 0,
+          properties: {
+            topicAliasMaximum: 3
+          }
+        })
+      })
+      serverClient.on('publish', function (packet) {
+        switch (publishCount++) {
+          case 0:
+            assert.strictEqual(packet.topic, 'test1')
+            assert.strictEqual(packet.properties.topicAlias, 1)
+            break
+          case 1:
+            assert.strictEqual(packet.topic, 'test2')
+            assert.strictEqual(packet.properties.topicAlias, 2)
+            break
+          case 2:
+            assert.strictEqual(packet.topic, 'test3')
+            assert.strictEqual(packet.properties.topicAlias, 3)
+            break
+          case 3:
+            assert.strictEqual(packet.topic, '')
+            assert.strictEqual(packet.properties.topicAlias, 1)
+            break
+          case 4:
+            assert.strictEqual(packet.topic, '')
+            assert.strictEqual(packet.properties.topicAlias, 3)
+            break
+          case 5:
+            assert.strictEqual(packet.topic, 'test4')
+            assert.strictEqual(packet.properties.topicAlias, 2)
+            server103.close()
+            client.end(true, done)
+            break
+        }
+      })
+    }).listen(ports.PORTAND103)
+
+    client.on('connect', function () {
+      // register topicAlias
+      client.publish('test1', 'Message')
+      client.publish('test2', 'Message')
+      client.publish('test3', 'Message')
+
+      // use topicAlias
+      client.publish('test1', 'Message')
+      client.publish('test3', 'Message')
+
+      // renew LRU topicAlias
+      client.publish('test4', 'Message')
```

Comment:
so here the topic alias will be updated dynamically because autoAssignTopicAlias is used?

Reply:
Yes.